test/SimpleWebRequest.spec.ts

```diff
@@ -92,7 +106,158 @@ describe('SimpleWebRequest', () => {
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            // first the critical one gets sent
+            expect(onSuccess3).toHaveBeenCalled();
+
+            // then the high, which was returned to the queue at after getting unblocked
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccess1).toHaveBeenCalled();
+
+            // and the low priority one gets sent last 
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccess2).toHaveBeenCalled();
+        });
+        
+        it('checks the blocked function again, once the request is on top of the queue', () => {
+            const url = faker.internet.url();
+            const method = 'GET';
+            const blockDefer = SyncTasks.Defer<void>();
+            const onSuccess1 = jasmine.createSpy('onSuccess1');
+            const onSuccess2 = jasmine.createSpy('onSuccess2');
+            const onSuccess3 = jasmine.createSpy('onSuccess3');
+            const blockSpy = jasmine.createSpy('blockSpy').and.callFake(() => blockDefer.promise());
+
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, blockSpy).start().then(onSuccess1);
+            jasmine.clock().tick(10);
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccess2);
+            jasmine.clock().tick(10);
+
+            SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
+            // add a new request to kick the queue
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccess3);
+            
+            expect(blockSpy).toHaveBeenCalled();
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccess2).toHaveBeenCalled();
+
+            // unblock the request
+            blockDefer.resolve(void 0);
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccess3).toHaveBeenCalled();
+
+            // check if the request at the top of the queue got called again
+            expect(blockSpy).toHaveBeenCalledTimes(2);
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccess1).toHaveBeenCalled();
+        });
+
+        it('fails the request, if the blocking promise rejects', done => {
+            SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
+            const url = faker.internet.url();
+            const method = 'GET';
+            const blockDefer = SyncTasks.Defer<void>();
+            const onSuccess1 = jasmine.createSpy('onSuccess1');
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, () => blockDefer.promise()).start().then(onSuccess1).catch(err => {
+                expect(1).toBe(1);
```

Comment:
Suggestion: Expect `onSuccess1` to have not been called, instead of `1 to be 1`.

Reply:
Totally forgot about this :)